boa/explorer.py

```diff
@@ -1,14 +1,29 @@
-import json
 import time
 from typing import Optional
 
-import requests
+from boa.rpc import json
 
-SESSION = requests.Session()
+try:
+    from requests_cache import CachedSession, Response
+
+    def _filter_fn(res: Response) -> bool:
+        data = res.json()
+        return _is_success_response(data) and not _is_rate_limited(data)
+
+    SESSION = CachedSession("etherscan_cache", filter_fn=_filter_fn)
```

Comment:
i think we want to add an expiry here (1 week), or maybe `cache_control=True` will be sufficient.

Reply:
from @romanagureev:
`allowable_codes=[200], filter_fn=lambda response: int(response.json().get("status", 0)) == 1`

Reply:
i just looked at the headers that etherscan returns, it sets `cache-control: private`, here is an example:
```
< HTTP/2 200 
< server: nginx
< date: Fri, 30 Aug 2024 10:19:57 GMT
< content-type: application/json; charset=utf-8
< content-length: 11343
< cache-control: private
< access-control-allow-origin: *
< access-control-allow-headers: Content-Type
< access-control-allow-methods: GET, POST, OPTIONS
< x-frame-options: SAMEORIGIN
```

Reply:
yea after checking the requests-cache docs i guess for the expiry thing we want to set both `expire_after=3600*24*7` and `cache_control=True`.

https://requests-cache.readthedocs.io/en/stable/user_guide/expiration.html#precedence